### PR TITLE
Beam refactoring - definer and size  (former Aperture as NState)

### PR DESCRIPTION
**Pull request by beteva**
Inherit from AbstractNState.
Using the new beam definer  #872 from @meguiraun

### CODE REVIEW

**Comment by beteva** [comment]
@meguiraun I've started using your beam_definer branch. For time being this is only aperture and slits, Please, have a look.
The definer will follow soon.

**Comment by marcus-oscarsson** [comment]
@beteva, @meguiraun Nice :). @meguiraun you can go ahead and merge this if you are happy with it. It would probably be nice with some additional documentation for `AbstractBeam` as its quite complext, or do you think that this is enough:
https://mxcubecore.readthedocs.io/en/stable/dev/abstract_classes.html#abstractbeam ?

**Comment by beteva** [comment]
The definer part (based on @meguiraun beam_definer branch) has been added. Some methods to handle the beam size have been added.

**Comment by marcus-oscarsson** [comment]
@beteva and @meguiraun are you happy with this as it is ?

**Comment by meguiraun** [comment]
I am happy with this. I still need to adapt to our custom hwobj but I believe it is a solid foundation.

File: mxcubecore/HardwareObjects/mockup/MotorMockupBis.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+"""
+
+from mxcubecore.HardwareObjects.abstract.AbstractMotor import AbstractMotor
+from mxcubecore.BaseHardwareObjects import HardwareObjectState
+
+__copyright__ = """ Copyright © 2019 by the MXCuBE collaboration """
+__license__ = "LGPLv3+"
+
+
+class MotorMockupBis(AbstractMotor):
```
Comment (meguiraun) [inline mxcubecore/HardwareObjects/mockup/MotorMockupBis.py:36]: why do we need a new mockup?

**Review by meguiraun — APPROVED** [review]

File: mxcubecore/HardwareObjects/mockup/MotorMockupBis.py (hunk 1 above)
Comment (beteva) [reply mxcubecore/HardwareObjects/mockup/MotorMockupBis.py:36]: I did this to be able to define two simple motors without breaking anybody's code. Probably it is worth looking more carefully what to be implemented and merge the two in one.

**Pull request merged**